src/Compiler/CompilerOptions.hpp

```diff
@@ -161,10 +173,14 @@ std::string getCompilerOption(const onnx_mlir::OptionKind kind);
 std::vector<std::string> getCompilerConfig(std::string k);
 void addCompilerConfig(std::string k, std::vector<std::string> v);
 void delCompilerConfig(std::string k, std::vector<std::string> v);
-// CompilerConfig may be set at initialization or inside CompileModule().
-// Since CompileModule() may be called repeated, push is used to mark
-// the current config and pop is used to restore the previously mark config
-void pushCompilerConfig(std::string k);
-void popCompilerConfig(std::string k);
 
+// Functions related to initializing compiler configuration states based on
+// parsed options
+std::optional<std::string> getEnvVar(std::string name);
+std::string getExecPath(void);
```

Comment:
nit: we generally omit `void` from C++ function signatures without arguments (try `grep -Er '^[a-zA-Z].*\((void)?\);$' src`)

Reply:
Then I suggest we change all function prototypes without arguments to `func(void)`.

`func(void)` and `func()` mean different things. `func(void)` means the function takes no argument. `func()` means there is no information about the arguments so you can call `func(a, b, c)` and the compiler will not generate a warning.

Reply:
no, that's C and not C++ https://stackoverflow.com/a/51080

in C++ it's most standard to use `()` to mean no arguments

Reply:
Quote from the link you provided:

By writing foo(void), therefore, we achieve the same interpretation across both languages and make our headers multilingual (though we usually need to do some more things to the headers to make them truly cross-language; namely, wrap them in an extern "C" if we're compiling C++).

So to me, even though the two are the same in C++, I personally would still use `func(void)`. I don't why MLIR does it the other way but fine I will change it.